packages/vkui/src/components/BaseGallery/CarouselBase/CarouselBase.tsx

```diff
@@ -0,0 +1,561 @@
+    onChange?.(
+      (slideIndex - 1 + layoutState.current.slides.length) % layoutState.current.slides.length,
+    );
+    onPrevClick?.(event);
+  };
+
+  const slideRight = (event: React.MouseEvent) => {
+    onChange?.((slideIndex + 1) % layoutState.current.slides.length);
+    onNextClick?.(event);
+  };
+
+  /*
+   * Получает индекс слайда, к которому будет осуществлен переход
+   */
+  const getTarget = (e: TouchEvent) => {
+    const expectDeltaX = (shiftXDeltaRef.current / e.duration) * 240 * 0.6;
+    const shift =
+      shiftXCurrentRef.current +
+      shiftXDeltaRef.current +
+      expectDeltaX -
+      (layoutState.current.max ?? 0);
+    const direction = shiftXDeltaRef.current < 0 ? 1 : -1;
+
+    // Находим ближайшую границу слайда к текущему отступу
+    let targetIndex = layoutState.current.slides.reduce(
+      (val: number, item: GallerySlidesState, index: number) => {
+        const previousValue = Math.abs(layoutState.current.slides[val].coordX + shift);
+        const currentValue = Math.abs(item.coordX + shift);
+
+        return previousValue < currentValue ? val : index;
+      },
+      slideIndex,
+    );
+
+    if (targetIndex === slideIndex) {
+      let targetSlide = slideIndex + direction;
+
+      if (targetSlide >= 0 && targetSlide < layoutState.current.slides.length) {
+        if (
+          Math.abs(shiftXDeltaRef.current) >
+          layoutState.current.slides[targetSlide].width * 0.05
+        ) {
+          targetIndex = targetSlide;
+        }
+      }
+      return direction < 0
+        ? (targetSlide + layoutState.current.slides.length) % layoutState.current.slides.length
+        : targetSlide % layoutState.current.slides.length;
+    }
+
+    return targetIndex;
+  };
+
+  const onStart = (e: TouchEvent) => {
+    onDragStart?.(e);
+    shiftXCurrentRef.current = calculateIndent(slideIndex);
+    shiftXDeltaRef.current = 0;
+  };
+
+  const onMoveX = (e: TouchEvent) => {
+    if (isDraggableProp && !layoutState.current.isFullyVisible) {
+      e.originalEvent.preventDefault();
+
+      if (e.isSlideX) {
+        if (shiftXDeltaRef.current !== e.shiftX) {
+          shiftXDeltaRef.current = e.shiftX;
+          requestTransform(shiftXCurrentRef.current + shiftXDeltaRef.current);
+        }
+      }
+    }
+  };
+
+  const onEnd = (e: TouchEvent) => {
+    const targetIndex = e.isSlide ? getTarget(e) : slideIndex ?? 0;
+    onDragEnd?.(e, targetIndex);
+
+    const shiftXStick = shiftXCurrentRef.current + shiftXDeltaRef.current;
+    if (targetIndex !== slideIndex) {
+      shiftXCurrentRef.current = shiftXStick;
+    }
+
+    if (targetIndex !== slideIndex) {
+      onChange?.(targetIndex);
+    } else {
+      const requiredShiftX = calculateIndent(targetIndex);
+      requestTransform(requiredShiftX, true);
+    }
+  };
+
+  const setSlideRef = (slideRef: HTMLDivElement | null, slideIndex: number) => {
+    slidesStore.current[`slide-${slideIndex}`] = slideRef;
```

Comment:
```suggestion
    slidesStore.current[slideIndex] = slideRef;
```
вижу можно без конкатенации